include/StochasticNoiseSimulator.hpp

```diff
@@ -3,189 +3,142 @@
-        initializeNoiseProbabilities(noise_prob);
-        setNumberOfRuns(stoch_runs);
-    }
+                             const std::string&                       cNoiseEffects,
+                             double                                   cNoiseProbability,
+                             std::optional<double>                    cAmpDampingProbability,
+                             double                                   cMultiQubitGateFactor,
+                             std::size_t                              cStochRuns,
+                             const std::string&                       recordedProperties,
+                             bool                                     unoptimizedSim,
+                             unsigned int                             stepNumber,
+                             double                                   stepFidelity,
+                             std::size_t                              seed = 0U):
+        Simulator<DDPackage>(seed),
+        qc(qc), stepNumber(stepNumber), stepFidelity(stepFidelity) {
+        // setNoiseEffects
+        for (const auto noise: cNoiseEffects) {
+            switch (noise) {
+                case 'A':
+                    noiseEffects.push_back(dd::amplitudeDamping);
+                    break;
+                case 'P':
+                    noiseEffects.push_back(dd::phaseFlip);
+                    break;
+                case 'D':
+                    noiseEffects.push_back(dd::depolarization);
+                    break;
+                default:
+                    throw std::runtime_error("Unknown noise operation '" + cNoiseEffects + "'\n");
+            }
+        }
 
-    std::map<std::string, std::size_t> Simulate(unsigned int shots) override;
+        // initializeNoiseProbabilities
+        noiseProbability = cNoiseProbability;
+        // Default value for amplitude damping prob is double the general error probability
+        amplitudeDampingProb = (cAmpDampingProbability) ? cAmpDampingProbability.value() : noiseProbability * 2;
+        multiQubitGateFactor = cMultiQubitGateFactor;
 
-    std::map<std::string, double> StochSimulate();
+        if (noiseProbability < 0 || amplitudeDampingProb < 0 || noiseProbability * multiQubitGateFactor > 1 || amplitudeDampingProb * multiQubitGateFactor > 1) {
+            throw std::runtime_error("Error probabilities are faulty!"
+                                     "\n single qubit error probability: " +
+                                     std::to_string(cNoiseProbability) +
+                                     " multi qubit error probability: " + std::to_string(cNoiseProbability * multiQubitGateFactor) +
+                                     "\n single qubit amplitude damping  probability: " + std::to_string(amplitudeDampingProb) +
+                                     " multi qubit amplitude damping  probability: " + std::to_string(amplitudeDampingProb * multiQubitGateFactor));
+        }
```

Comment:
i would assume this to be the same error message as in the deterministic part -> extract into a function and reuse.